File: AI_BASIC/AI_BASIC/VirtualMachine/SAL_ZX21_VM_UI/SAL_ZX21_ConsoleUI.cs
using System;
using SAL_VM.STACK_VM;

namespace SAL_VM
{
    public partial class SAL_ZX21_ConsoleUI
    {
        public SAL_ZX21_ConsoleUI()
        {
            InitializeComponent();
            _ButtonRunCode.Name = "ButtonRunCode";
            _ButtonClrScrn.Name = "ButtonClrScrn";
            _ButtonNewScrn.Name = "ButtonNewScrn";
        }

        public string ExecuteCode(ref string Code)
        {
            return X86API.RunMachineCode(ref Code);
        }

        private void ButtonRunCode_Click(object sender, EventArgs e)
        {
            string CodeAnalysis = RichTextBoxCodeEntry.Text;
            RichTextBoxInfo.Text += ExecuteCode(ref CodeAnalysis);
        }

        private void ButtonNewScrn_Click(object sender, EventArgs e)
        {
            var frm = new SAL_ZX21_ConsoleUI();
            frm.Show();
        }

        private void ButtonClrScrn_Click(object sender, EventArgs e)
        {
            RichTextBoxInfo.Clear();
        }
    }
}

File: SpydazWeb AI Virtual Machine/SourceCode/SAL_ZX21_VM_UI/SAL_ZX21_ConsoleUI.vb
Imports SAL_VM.STACK_VM

Public Class SAL_ZX21_ConsoleUI

    Public Function ExecuteCode(ByRef Code As String) As String
        Return X86API.RunMachineCode(Code)
    End Function

    Private Sub ButtonRunCode_Click(sender As Object, e As EventArgs) Handles ButtonRunCode.Click
        Dim CodeAnalysis As String = RichTextBoxCodeEntry.Text
        RichTextBoxInfo.Text &= ExecuteCode(CodeAnalysis)
    End Sub

    Private Sub ButtonNewScrn_Click(sender As Object, e As EventArgs) Handles ButtonNewScrn.Click
        Dim frm As New SAL_ZX21_ConsoleUI
        frm.Show()
    End Sub

    Private Sub ButtonClrScrn_Click(sender As Object, e As EventArgs) Handles ButtonClrScrn.Click
        RichTextBoxInfo.Clear()
    End Sub
End Class